mri/operators/proximity/ordered_weighted_l1_norm.py

```diff
@@ -105,36 +129,65 @@ def _op_method(self, data, extra_factor=1.0):
         data: np.ndarray
             Input array of data
         """
-        if self.mode is 'all':
+        if self.mode == 'all':
             output = np.reshape(
                 self.owl_operator.op(data.flatten(), extra_factor),
                 data.shape
             )
             return output
-        elif self.mode is 'band_based':
+        elif self.mode == 'band_based':
             data_r = self._reshape_band_based(data)
             output = Parallel(n_jobs=self.n_jobs)(
                 delayed(self.owl_operator[i].op)(
                     data_band,
-                    extra_factor)
-                for i, data_band in enumerate(data_r))
+                    extra_factor
+                )
+                for i, data_band in enumerate(data_r)
+            )
             reshaped_data = np.zeros(data.shape, dtype=data.dtype)
             start = 0
             n_channel = data.shape[0]
-            for band_shape_idx, band_data in zip(self.band_shape, output):
-                stop = start + np.prod(band_shape_idx)
+            for band_size, band_data in zip(self.band_sizes, output):
+                stop = start + band_size
                 reshaped_data[:, start:stop] = np.reshape(
                     band_data,
-                    (n_channel, np.prod(band_shape_idx)))
+                    (n_channel, band_size)
+                )
+                start = stop
+            output = np.asarray(reshaped_data).T
+            return np.asarray(output).T
+        elif self.mode == 'scale_based':
```

Comment:
I feel that there is a lot of mutualisation that could happen between `scale_based` and `band_based`,
If I understand what they are correctly, then it's only normal that this would happen, and I think it should be reflected by the code.

The only difference is more or less the initial reshaping of the data. Even the final one can be largely refactored to be shared.

(Maybe even the `coef_based` could be included but I don't want to get ahead of myself).

Reply:
Ahhh, i get your point, ill see what i can do.

Reply:
@zaccharieramzi I have made changes to tackl;e this redudancy, but sure made the code a little more complicated, do let me know if it is harder to understand.